core/src/test/java/io/fabric8/maven/core/service/kubernetes/JibBuildServiceTest.java

```diff
@@ -82,9 +81,8 @@
             .buildConfig(buildImageConfiguration)
             .build();
 
-    @Test
+    @Test(expected = IllegalStateException.class)
```

Comment:
Doesn't make much sense to me for a test named `testSucessful...`, also, I guess that some assertions within the code block must be unreachable.

Reply:
Shall I remove this test?

Reply:
Probably some parts of the test should be updated. The original version of the test was there for some reason `testSuccessfulBuildOffline` (but given the fact that the next test `testSuccessfulBuild` also throws an exception - who knows...)

Haven't checked the code yet, but I think some mocking should be added to the expectations block.

Reply:
tbh, I don't like testing this functionality in a unit test. Otherwise I can remove unreachable assertions and rename test also

Reply:
I want to check the PR on my machine first thing tomorrow, so I can understand how this thing works. Haven't had the time yet to play with the JIB implementation.

Reply:
Actually for offline build, you need images cached in your `~/.cache/google-cloud-tools/jib` directory. And for successful build(which includes a push to docker registry), we need dockerhub credentials set up. Not sure how to mock this.

Reply:
Sure, np

Reply:
If your implementation is working, disable the tests if you want for the current PR.
I can try to update them in a new PR

Reply:
I've disabled these two tests for now. You can take a look at this demo project of mine to test the implementation: https://github.com/r0haaaan/fmp-jib-helloworld-sample